src/roswire/ros2/reader.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+import json
+import os
+import shlex
+import typing
+from typing import Any, Dict, Optional, Sequence
+
+import attr
+import pkg_resources
+from loguru import logger
+
+from ..proxy.roslaunch.config import ExecutableType, LaunchConfig, NodeConfig
+from ..proxy.roslaunch.reader import LaunchFileReader
+
+if typing.TYPE_CHECKING:
+    from .. import AppInstance
+
+
+@attr.s(auto_attribs=True)
+class ROS2LaunchFileReader(LaunchFileReader):
+
+    @classmethod
+    def for_app_instance(cls,
+                         app_instance: 'AppInstance'
+                         ) -> 'LaunchFileReader':
+        return ROS2LaunchFileReader(app_instance)
+
+    _app_instance: 'AppInstance'
```

Comment:
Attributes should go to the top of the class definition (i.e., line 21).

Reply:
Done